client/components/async-load/index.jsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { omit } from 'lodash';
+
+export default class AsyncLoad extends Component {
+	static propTypes = {
+		require: PropTypes.func.isRequired,
+		placeholder: PropTypes.node
+	};
+
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			component: null
+		};
+	}
+
+	componentWillMount() {
```

Comment:
This brings to mind another Babel plugin that might be interesting to experiment with:

http://babeljs.io/docs/plugins/transform-react-constant-elements/